app/src/main/java/com/igordotsenko/dotsenkorssreader/ItemListActivity.java

```diff
@@ -19,290 +15,165 @@
-                if ( intent.getAction().equals(RefreshService.AUTOREFRESH_RESULT_ACTION + currentChannelId) ) {
-                    handleAutoRefreshResult(intent);
-                }
-            }
-        };
-
-        //Intent filter for interacting with RefreshService initialization
-        IntentFilter filter = new IntentFilter();
-        filter.addAction(RefreshService.AUTOREFRESH_RESULT_ACTION + currentChannelId);
-        filter.addAction(RefreshService.AUTOREFRESH_MESSAGE_ACTION + currentChannelId);
-        localBroadcastManager.registerReceiver(broadcastReceiver, filter);
-
-        //Start RefreshSercive
-        Intent intent = new Intent(ItemListActivity.this, RefreshService.class);
-        startService(intent.putExtra(Channel.ID, currentChannelId));
+        mCurrentChannelId = getIntent().getLongExtra(Channel.ID, -1);
 
         //SwipeRefreshLayout initialization
-        swipeRefreshLayout = (SwipeRefreshLayout) findViewById(R.id.item_list_swiperefresh_layout);
-        swipeRefreshLayout.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
+        mSwipeRefreshLayout = (SwipeRefreshLayout) findViewById(R.id.item_list_swiperefresh_layout);
+        mSwipeRefreshLayout.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
             @Override
             public void onRefresh() {
-                Handler handler = new Handler(Looper.getMainLooper()) {
-                    @Override
-                    public void handleMessage(Message message) {
-                        super.handleMessage(message);
-                        Toast.makeText(getBaseContext(), message.getData().getString(RefreshingRunnable.MESSAGE_TAG), Toast.LENGTH_SHORT).show();
-                        updateItemList();
-                        swipeRefreshLayout.setRefreshing(false);
-                    }
-                };
-
-                Thread thread = new Thread(new RefreshingRunnable(handler.obtainMessage()));
-                thread.start();
+                ContentResolver.requestSync(
+                        MainActivity.sAccount,
+                        ReaderContentProvider.ContractClass.AUTHORITY,
+                        Bundle.EMPTY);
+                mSwipeRefreshLayout.setRefreshing(false);
             }
         });
 
         //SearchView initialization
-        searchView = (SearchView) findViewById(R.id.item_list_search_view);
-        searchView.setOnQueryTextListener(this);
-        searchView.setIconifiedByDefault(false);
+        mSearchView = (SearchView) findViewById(R.id.item_list_search_view);
+        mSearchView.setOnQueryTextListener(this);
+        mSearchView.setIconifiedByDefault(false);
 
         //RecyclerView initialization
-        recyclerView = (RecyclerView) findViewById(R.id.item_list_recyclerview);
+        mRecyclerView = (RecyclerView) findViewById(R.id.item_list_recyclerview);
         LinearLayoutManager llm = new LinearLayoutManager(ItemListActivity.this);
```

Comment:
Fixed this issue, but as I understand changes are not tracked automatically because comment is set on another line )